### ISSUE TITLE
Incompatible with Pydantic 2

### ISSUE THREAD

**Issue by prescod** [Ovsyanka83/pydantic-duality#14]
Any attempt to import `DualBaseModel` into application code results in:

```
     testpd.py", line 3, in <module>
    from pydantic_duality import DualBaseModel
  File "site-packages/pydantic_duality/__init__.py", line 22, in <module>
    from pydantic.main import ModelMetaclass
ImportError: cannot import name 'ModelMetaclass' from 'pydantic.main' 
```
Using:
```
pydantic                                 2.7.4
pydantic_core                            2.18.4
pydantic-duality                         1.2.0
pydantic-settings                        2.3.2
```

**Comment by zmievsa** [comment]
Yup! I'm adding the compatibility soon but it's not there yet.

**Comment by zmievsa** [comment]
Released in pydantic-duality==2.0.0. Please, have a look
